### PR TITLE
feat(wezterm-gui): Support move to word end

### PR DESCRIPTION
**Pull request by GZLiew**
Add support for vim-like navigation `e` key where the cursor jumps to end of the word

https://user-images.githubusercontent.com/43195293/211966200-439b6dd3-b336-4ebf-a5cb-72a32d085664.mov

### CODE REVIEW

**Comment by wez** [comment]
Thanks for this!
Could you make a separate method for this new movement style?
I think it would result in a less complex method overall.
Thanks!

**Comment by GZLiew** [comment]
@wez just noticed `move_forward_one_word` doesnt jump to the next line as well. Do you want me to create a pr for that?

**Comment by wez** [comment]
When I use `w` to move forward a word, it will wrap/jump to the next line. What's not working for you there?

**Comment by GZLiew** [comment]
I created a pr with a video showing the problem 😄  #2955

**Comment by GZLiew** [comment]
@wez is this pr good enough?

File: docs/copymode.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ reassignable.
 |                       | `alt-b`         |
 |                       | `Shift-Tab`     |
 |                       | `b`             |
+| Move forward one word end    | `e` |
```
Comment (wez) [inline docs/copymode.md:55]: Please rebase and amend this to match the new style used in this doc:

```suggestion
| Move forward one word end    | <kbd>e</kbd> |
```

**Review by wez — APPROVED** [review]
Looks good; please rebase and fixup the copymode docs and I'll merge it!

File: docs/copymode.md (hunk 1 above)
Comment (GZLiew) [reply docs/copymode.md]: thanks @wez!

**Comment by wez** [comment]
Ah, looks like the merge commit you made makes your fork into something that cannot be rebased onto my fork any more because of the diamond graph.  I only allow rebase merging in this repo to keep the commit graph linear.

Can you fix up your fork/PR so that the branch you want to use in your PR is my `main` with your commits on top, and no merge commits?  Then I can merge it.  Thanks!

**Comment by wez** [comment]
Thanks!

**Pull request merged**